data/en/generatepbkdfkey.json

```diff
@@ -1,25 +1,25 @@
 {
 	"name":"generatePBKDFKey",
-	"syntax":"generatePBKDFKey(Algorithm, Passphrase, Salt, Iterations, KeySize)",
+   	"syntax":"generatePBKDFKey(algorithm,passphrase,salt);",
 	"returns":"string",
 	"description":"CFML implementation of Password-Based Key-Derivation Function (PBKDF).",
 	"type":"function",
-	"related":["hash", "encrypt","decrypt"],
+	"related":["hash", "encrypt","decrypt","generateSecretKey"],
+	"params":[
+        {"name":"algorithm", "description":"Hashing algorithm used for generating key", "required":true, "default":"", "type":"string", "values":["PBKDF2WithHmacSHA1","PBKDF2WithSHA1","PBKDF2WithSHA224","PBKDF2WithSHA256","PBKDF2WithSHA384","PBKDF2WithSHA512"]},
+        {"name":"passphrase", "description":"Passphrase used for the key. KEEP THIS SECRET.", "required":true, "default":"", "type":"string", "values":[]},
+        {"name":"salt", "description":"A string which will be added to the passphrase before encryption", "required":true, "default":"", "type":"string", "values":[]},
+        {"name":"iterations", "description":"The number of PBKDEF iterations to perform. A minimum recommended value is 1000", "required":false, "default":"", "type":"numeric", "values":[]},
+        {"name":"keysize", "description":"The length in bytes of the key to generate", "required":false, "default":"", "type":"numeric", "values":["true", "false"]}
+    ],
 	"engines":{
-		"coldfusion":{"minimum_version":"11","docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-e-g/generatepbkdfkey.html","notes":""},
-		"lucee":{"minimum_version":"5","docs":"http://docs.lucee.org/reference/functions/generatepbkdfkey.html","notes":""}
+		"coldfusion":{"minimum_version":"11","docs":"https://helpx.adobe.com/coldfusion/cfml-reference/coldfusion-functions/functions-e-g/generatepbkdfkey.html","notes":"These algorithms are available only in enterprise versions:\n\nPBKDF2WithSHA1\nPBKDF2WithSHA224\nPBKDF2WithSHA256\nPBKDF2WithSHA384\nPBKDF2WithSHA512\nPBKDF2WithSHA512-224\nPBKDF2WithSHA512-256"},
+		"lucee":{"minimum_version":"5","docs":"http://docs.lucee.org/reference/functions/generatepbkdfkey.html","notes":"Lucee seems to support only \"PBKDF2WithHmacSHA1\" and \"PBKDF2WithHmacSHA512\" are allowed values for algorithm argument"}
```

Comment:
I mean all PBKDF2WithHmacSHA are working from 1 to 512 (so also 224, 256 etc).
But all PBKDF2WithSHA are not working.

Reply:
In Lucee or Adobe ColdFusion?

Reply:
According to your previous comment, I guess it's Lucee^^

Reply:
@Xcreen @shaedrich Adobe ColdFusion Enterprise includes a java crypto provider that implements these algorithms, for Lucee it is up to the provider that you have installed, if using the default java crypto provider it only supports `PBKDF2WithHmacSHA1` on Java 1.7 for example. If you are using Java 8 it supports more algorithms. 

So it is not that Lucee does not support the algorithms, support depends on what crypto provider you have.

Reply:
Okay, I add this.

Reply:
Looking at it further, I see you were talking about the non HMAC version of the algorithms. Still it is up to the provider, but it seams the default JCE does not implement the non hmac algorithms at this time. We should add `PBKDF2WithHmacSHA256` `PBKDF2WithHmacSHA512` `PBKDF2WithHmacSHA384` `PBKDF2WithHmacSHA224` and to the algorithm `values`.

Reply:
It would be good to add something like this to the examples, this will list all valid PBKDF Algorithms on your current engine:

```
for (provider in createObject("java", "java.security.Security").getProviders()) {
  for (service in provider.getServices().toArray()) {
    if (service.getAlgorithm() contains "PBKDF") {
      writeOutput("<h2>#service.getAlgorithm()#</h2><p>#service.toString()#</p>");
    }
  }
}
```

https://trycf.com/gist/5d3f6309803b115f74388b0ef7579b75/lucee5

Reply:
In my opinion, your function is not part of the doc for that specific function. Adding it to a guide would be better, I guess.

So you can add the non HMAC version in Adobe ColdFision as well? In that case there would not be the need for the enterprise edition.